doc: add control plane configuration for Azure private location (#141)
gatling/frontline-cloud-doc: content/reference/admin/private_locations/installation/azure/index.md

{{< alert warning >}}
Container App creation steps on Azure portal do not allow to configure everything you need, like the File Share mounting. Please, take care to go through all steps for proper configuration.
{{< /alert >}}

### Control plane application creation

On your Azure portal, click on "Create a resource" button, and choose "Container Apps".

{{< img src="azure-cp-installation-create-resource.png" alt="Create resource" >}}

Start by referencing basic information about your app, as requested by the wizard.

You can configure freely the subscription, group, etc. at this step, according to your policies.

{{< img src="azure-cp-installation-create-app1.png" alt="Create application step 1" >}}

You will then need to configure the Gatling control plane container as the base image of the application:
* Go to "App settings tab"
* Uncheck "Use quickstart image"
* Configure a container name as you desire
* Choose "Docker Hub or other registries" as image source
* Choose "Public" as image type
* Indicate "docker.io" in registry login server field

> * Indicate "gatlingcorp/control-plane:latest" in image and tag field

Reviewer: ```suggestion
* Indicate `gatlingcorp/control-plane:latest` in image and tag field
```

* Navigate to your application page on the portal
* Click on "Container" in the side menu (you can check their configuration while you are here)
* Click on "Edit and deploy" button in the top menu
{{< img src="azure-cp-installation-configure-container-file-share1.png" alt="Configure containers file share step 1" >}}

* In the "Container" tab, click on the existing container automatically created at application creation. Or, there is none, click on the "Add" button"
{{< img src="azure-cp-installation-configure-container-file-share2.png" alt="Configure containers file share step 2" >}}

* Go to "Volume mounts" tab
* Select your newly created file share in "File share name" drop box
* Enter "/app/conf" as "Mount path (folder)"
* Check next step before validating changes (additional configuration) since you can do it at the same time.
{{< img src="azure-cp-installation-configure-container-file-share3.png" alt="Configure containers file share step 3" >}}

### Additional containers configuration

During the application container configuration step, you can set scale rules for your application.

{{< alert warning >}}

> We strongly advise to set the minimum scale at 1, to ensure that your control plane will always be up.

Reviewer: ```suggestion
We strongly advise to set the minimum scale to 1, to ensure that your control plane will always be up.
```

{{< /alert >}}

* In the "Scale" tab, set the min and max instances with the ruler
* Set rules if you need to (not mandatory)
* Don't forget to validate your changes

{{< img src="azure-cp-installation-configure-container-scale.png" alt="Configure containers instances minimum scale" >}}

## Use your new control plane

Once you've configured your control plane container, it should automatically start a new revision with desired configuration.

You can check logs in your application "Log stream" menu, or perform more complex requests in "Logs" menu.
Be sure to select the currently active revision logs.

> Now that your control plane is up and running, after a short time you should see your control plane with "up" status in Gatling Enterprise Cloud.

Reviewer: ```suggestion
Now that your control plane is up and running, after a short time you should see your control plane with {{< badge success >}}up{{< /badge >}} status in Gatling Enterprise Cloud.
```

{{< img src="azure-cp-status.png" alt="Checking out the Control Plane's status in Gatling Enterprise Cloud" >}}

> You can now configure a simulation to run one one or more of this Control Plane's pool!

Reviewer: ```suggestion
You can now configure a simulation to run one or more of this Control Plane's pool!
```

---
title: "Azure Container App"
description: "How to install a Gatling Control Plane on Azure using Container Apps and Azure Files, to set up your Private Locations and run load generators in your own Azure network"

> lead: "Run a Control Plane on Azure using Container Apps and Fargate, to set up your Private Locations and run load generators in your own Azure network"

Reviewer: Indeed

> lead: "Run a Control Plane on Azure using Container Apps and Fargate, to set up your Private Locations and run load generators in your own Azure network"

Author: Are you REALLY sure ? 🫥

lastmod: 2023-03-04T16:00:00+00:00
weight: 22052
---

First of all, you should have followed [introduction](http://localhost:1313/docs/cloud/reference/admin/private_locations/introduction/) instructions to configure Gatling Cloud for receiving a new control plane. Please check this section first if this is not the case.

Azure [Container Apps](https://azure.microsoft.com/en-us/products/container-apps) is a managed serverless container services. You can use it to deploy your own control plane instances without creating dedicated virtual machines.

Keep in mind that this is only one way to install a control plane in Azure, one can use any other Azure service able to deploy a container.

## Azure services overview

Some Azure services are needed for this control plane installation:
- [Container Apps](https://azure.microsoft.com/en-us/products/container-apps): the base product for containerized serverless applications
- [Azure Files](https://azure.microsoft.com/en-us/products/storage/files): file share for configuration file storage

## Configuration file share

Control plane locations configuration is based on a file that has to be shared with the application.

> See [control plane configuration](/docs/cloud/reference/admin/private_locations/introduction/#configuration) sections of Gatling Cloud documentation for details.

Reviewer: ```suggestion
See [control plane configuration]({{< ref "../../introduction#configuration" >}}) sections of Gatling Cloud documentation for details.
```

In your storage account, let's create a File Share for the control plane configuration file:
* Access to your storage account page
* Click on "File shares" in the side menu
* Click on "File share" creation button in the top menu
* Configure and finalize creation

Write down the file share name, you will need it in further steps.

{{< img src="azure-cp-installation-create-file-share.png" alt="Create file share" >}}

## Container App

A Container App is what will manage the proper control plane application.

There are several ways of creating and configuring it, as described in [Azure Container Apps documentation](https://learn.microsoft.com/en-us/azure/container-apps/quickstart-portal). We are going to consider only the Azure portal way of doing it.

{{< alert warning >}}
Container App creation steps on Azure portal do not allow to configure everything you need, like the File Share mounting. Please, take care to go through all steps for proper configuration.
{{< /alert >}}

{{< alert info >}}

> Each configuration can be 

Author: Well, it shouldn't have been important, because I don't recall what I was going to say
